hydroflow/src/scheduled/graph.rs

```diff
@@ -93,6 +121,27 @@ impl Hydroflow {
         }
     }
 
+    /// Go to the next stratum which has work available. Return true if more work is available, otherwise false if no work is immediately available on any strata.
+    pub fn next_stratum(&mut self) -> bool {
+        self.try_recv_events();
+
+        let old_stratum = self.current_stratum;
+        while {
+            self.current_stratum += 1;
+            if self.current_stratum >= self.ready_queue.len() {
+                self.current_stratum = 0;
+                self.current_epoch += 1;
+            }
+            old_stratum != self.current_stratum
+        } {
+            if !self.ready_queue[self.current_stratum].is_empty() {
+                return true;
+            }
+        }
```

Comment:
Ok, this loop is pretty janky, I will fix it

Reply:
Technically has a bug, if the only stratum which has work is the current stratum it will return false.